### PR TITLE
IEN - 512 Job ID is nolonger mandatory

### PR DESCRIPTION
**Pull request by dbayly-freshworks**
Removed the `not empty` flag for Job Id 
Added UUID on before insert 
Also tweeked package log to accept more recent versions of node. My docker got wiped out and the current version of Node-alpine -16 will have Node v 16.16.0 installed (which will cause the application to exit because it's not 16.15.1.

### CODE REVIEW

**Review by bhumin-fw — APPROVED** [review]

File: apps/api/src/applicant/entity/ienjob.entity.ts [hunk 1]
```diff
@@ -66,4 +68,10 @@ export class IENApplicantJob {
       this.status_audit = sortStatus(this.status_audit);
     }
   }
+  @BeforeInsert()
+  beforeInsert() {
```
Comment (sunghwan-park-fw) [inline apps/api/src/applicant/entity/ienjob.entity.ts:72]: this job_id is for reference to a real job posting. so I don't think it makes sense to generate a random id.

Comment (dbayly-freshworks) [reply apps/api/src/applicant/entity/ienjob.entity.ts:72]: I think having this field blank and not generating one will create more issues than it solves. 
If we don't have it, we'll have trouble differentiating between job competitions. Also, we'll have an awkward edge case when checking for the uniqueness of the ID field. 

Plus we'll need to sync jobs with ATS eventually and having that uuid will allow that process to go smoothly. 

The main case I can against it is in the case of de-duplication or matching with an existing job within ATS. I don't think that's within the scope of our work.

Comment (sunghwan-park-fw) [reply apps/api/src/applicant/entity/ienjob.entity.ts:72]: I get your point. Another question. randomUUID is a little long and verbose to display. Do we want to abbreviate it, like the first 4 bytes?

**Review by sunghwan-park-fw — APPROVED** [review]

File: apps/api/src/applicant/entity/ienjob.entity.ts (hunk 1 above)
Comment (dbayly-freshworks) [reply apps/api/src/applicant/entity/ienjob.entity.ts:72]: That works 💯

**Pull request merged**